### DISCUSSION TITLE
How to migrate to another VPS

### DISCUSSION THREAD

**Discussion by akondratiuk1** [Dokploy/dokploy#369] [Q&A] [answered]
I want to migrate to another VPS and keep all my project and their settings.
for example, I use my app, grafana, prometheus
I want to backup:
1. app: environment, domain, and volume settings
2. grafana: all settings, alarms, dashboards, etc.. 
3. prometheus: collected metrics

What do you think is the best way to do it?

**Comment by rahnamaj22** [comment, marked as answer]
I've recently migrated to another server with the following services on Dokploy:
- Python application
- .NET backend
- Directus
- Appsmith
- Several Postgres and Mongo databases

Unfortunately, my VPS provider lacked backup capabilities, so I had to migrate everything manually. Here’s what I did:

- Got a new VPS
- Transferred the entire filesystem using rsync:
`rsync -aAXv --delete \
    --exclude={"/dev/*","/proc/*","/sys/*","/tmp/*","/run/*","/mnt/*","/media/*","/lost+found","/swapfile"} \
    -e "ssh -i /path/to/private_key" user@source_vps_ip:/ /
`
- After the migration, I updated the server IP in the Dokploy database:
`UPDATE admin SET "serverIp" = 'new_server_ip' WHERE "serverIp" = 'old_server_ip';
`

This allowed me to move everything from one server to another(Including Dokploy). The only issue I encountered was with Appsmith, which I couldn't get to work after the migration.

**Reply by bloor** [reply]
How do you access the Dokploy database. What engine does it use and where it is ?

**Reply by rahnamaj22** [reply]
Just SSH to your server or use dokploy dashboard and click on "Enter the terminal" in Settings --> Server and then enter: **docker ps** to see the containers. there should be a postgres database which Dokploy is using

**Reply by sorenkrt** [reply]
Great solution. I migrated today successfully. Thanks

**Reply by kiliangui** [reply]
The admin table don't exist anymore. The serverIp column has been moved to the webServerSettings table. You can use this command
```
UPDATE "webServerSettings"
SET "serverIp" = 'new_server_ip'
WHERE "serverIp" = 'old_server_ip';
```

To connect to the dokploy db, first log with bash in the postgres container (find the container id with "docker ps")
`docker exec -it #containerId# bash`
Then connect to the db : 
`psql -U dokploy`

**Comment by kdurek** [comment]
Related to update the server IP in https://github.com/Dokploy/dokploy/pull/714

**Comment by lucasctd** [comment]
In case somebody else faces this situation I gonna share what I did:

1- Installed Dokploy on the new server
2- Stopped Docker on the new server
3- Used SCP to copy `/etc/dokploy` content from old server to the new one
4- Used SCP to copy the content of the `dokploy-postgres-database`, `dokploy-docker-config` and `redis-data-volume` volumes to the respective volumes on the new server
5- Fixed volumes' content permissions (Postgress should be 999:999)
6- Restarted docker
7- Created a SSH Key (already using Dokploy's UI) and added it to the authorized list of my remote servers
8- Updated Server's IP using the dashboard

Everything worked as expected after that. Hope it helps. 😉

**Reply by moxplod** [reply]
Did you have any databases like Postgres for your apps - did they move correctly with this approach?

**Reply by 0x0elliot** [reply]
thanks!
